test/test.js

```diff
@@ -0,0 +1,141 @@
+const inv = windows.createWindow(2, 'minecraft:inventory', 'inventory')
+
+const createClick = (mode, mouseButton, slot, useInv = false) => ({
+  slot,
+  mouseButton,
+  mode,
+  windowId: useInv ? inv.id : chest.id,
+  item: slot === -999 ? null : useInv ? inv.slots[slot] : chest.slots[slot]
+})
+
+chest.updateSlot(0, new Item(1, 64))
+chest.updateSlot(1, new Item(1, 1))
+chest.updateSlot(3, new Item(1, 63))
+chest.updateSlot(5, new Item(2, 3))
+chest.updateSlot(chest.inventoryEnd - 1, new Item(1, 44))
+chest.updateSlot(chest.inventoryEnd - 2, new Item(1, 44))
+chest.updateSlot(chest.inventoryEnd - 3, new Item(1, 44))
+
+inv.updateSlot(inv.inventoryEnd - 1, new Item(301, 1))
+inv.updateSlot(inv.inventoryEnd - 2, new Item(1, 1))
+
+describe('mode 0 | normal click', () => {
+  describe('mouseButton 0', () => {
+    it('pickup item', () => {
+      chest.acceptClick(createClick(0, 0, 0))
+      assert(!chest.slots[0] && chest.selectedItem?.count === 64 && chest.selectedItem.type === 1)
+    })
+  })
+
+  describe('mouseButton 1', () => {
+    it('drop one of selected Item into a slot (same item)', () => {
+      chest.acceptClick(createClick(0, 1, 1))
+      assert(chest.selectedItem?.count === 63 && chest.slots[1]?.count === 2)
+    })
+    it('drop one of selected Item into a slot (empty slot)', () => {
+      chest.acceptClick(createClick(0, 1, 2))
+      assert(chest.selectedItem?.count === 62 && chest.slots[2]?.count === 1 && chest.slots[2].type === 1)
+    })
+  })
+
+  describe('mouseButton 0', () => {
+    it('drop all of selected Item into a slot (almost full with same item)', () => {
+      chest.acceptClick(createClick(0, 0, 3))
+      assert(chest.selectedItem?.count === 61 && chest.slots[3]?.count === 64 && chest.slots[3].type === 1)
+    })
+    it('drop selected Item into empty slot', () => {
+      chest.acceptClick(createClick(0, 0, 0))
+      assert(Boolean(!chest.selectedItem && chest.slots[0]))
+    })
+  })
+})
+
+describe('mode 1 | shift click', () => {
+  describe('mouseButton 0', () => {
+    it('shift out of chest into inventory', () => {
+      chest.acceptClick(createClick(1, 0, 0))
+      assert(chest.slots[chest.inventoryEnd - 1]?.count === 64 && chest.slots[chest.inventoryEnd - 2]?.count === 64 && chest.slots[chest.inventoryEnd - 3]?.count === 64 && chest.slots[chest.inventoryEnd - 4]?.count === 1)
+    })
+
+    it('shift out of inventory into chest', () => {
+      chest.acceptClick(createClick(1, 0, chest.inventoryEnd - 1))
+      assert(!chest.slots[0] && chest.slots[1]?.count === 64 && chest.slots[2]?.count === 3)
+    })
+
+    it('shift out of inventory into chest #2', () => {
+      chest.acceptClick(createClick(1, 0, chest.inventoryEnd - 2))
+      assert(!chest.slots[chest.inventoryEnd - 2] && chest.slots[2]?.count === 64 && chest.slots[0]?.count === 3)
+    })
+
+    it.skip('shift out of inventory into armor slot - leather boots', () => {
+      // isn't even implemented, so it would naturally fail.
```

Comment:
Is the implementation planed? Or is there an active issue for this?

Reply:
neither,
in the current prismarine-windows you'd have to explicitly trigger this behaviour.
i also couldn't really make up how to implement this as i didn't find any mcData stuff which describes what items can go into which slot

an issue should probably be created for this.